### PR TITLE
feat: CLB Excel导入-创建URL规则 --story=119589495

### PR DESCRIPTION
**Pull request by KooKouse**

### CODE REVIEW

File: cmd/cloud-server/logics/load-balancer/create_url_rule_executor.go [hunk 1]
```diff
@@ -0,0 +1,460 @@
...
+		logs.Errorf("check resource flow relation failed, err: %v, rid: %s", err, kt.Rid)
+		return "", err
+	}
+	flowID, err := c.createFlowTask(kt, lb.ID, result)
+	if err != nil {
+		logs.Errorf("create flow task failed, err: %v, rid: %s", err, kt.Rid)
+		return "", err
+	}
+	err = lockResFlowStatus(kt, c.dataServiceCli, c.taskCli, lb.ID,
+		enumor.LoadBalancerCloudResType, flowID, enumor.AddRSTaskType)
```
Comment (ZQHcode) [inline cmd/cloud-server/logics/load-balancer/create_url_rule_executor.go:212]: 是enumor.CreateUrlRuleTaskType吧

File: cmd/cloud-server/logics/load-balancer/create_url_rule_preview_executor.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+	if err != nil {
+		logs.Errorf("validate data failed, err: %v, rid: %s", err, kt.Rid)
+		return nil, err
+	}
+	return c.details, nil
+}
+
+func (c *CreateUrlRulePreviewExecutor) convertDataToPreview(rawData [][]string) error {
+	for _, data := range rawData {
+		data = trimSpaceForSlice(data)
```
Comment (ZQHcode) [inline cmd/cloud-server/logics/load-balancer/create_url_rule_preview_executor.go:73]: data需要坐下长度判断，防止下面panic

File: cmd/cloud-server/logics/load-balancer/validate.go [hunk 3]
```diff
@@ -61,18 +61,18 @@ func validateInstType(instType enumor.InstType) error {
 
 func validateWeight(weight int) error {
 	if weight < 0 || weight > 100 {
-		return errors.New("权重范围错误")
+		return errors.New("weight range has error")
 	}
```
Comment (chenjr15) [inline cmd/cloud-server/logics/load-balancer/validate.go:65]: invalid weight

File: cmd/cloud-server/logics/load-balancer/validate.go [hunk 4]
```diff
@@ -42,11 +42,11 @@ func validateScheduler(scheduler enumor.Scheduler) error {
...
 func validatePort(ports []int) error {
 	if len(ports) > 2 || len(ports) == 0 {
-		return errors.New("端口数量错误")
+		return errors.New("port count has error")
 	}
 	for _, port := range ports {
 		if port < 0 || port > 65535 {
-			return fmt.Errorf("端口范围错误: %d ", port)
+			return fmt.Errorf("port range has error: %d ", port)
 		}
```
Comment (chenjr15) [inline cmd/cloud-server/logics/load-balancer/validate.go:50]: invaild ports length
invaild port, out of range

File: cmd/cloud-server/logics/load-balancer/create_url_rule_preview_executor.go [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+			detail.DefaultDomain = true
+		case "否":
+			detail.DefaultDomain = false
+		default:
+			return fmt.Errorf("DefaultDomain: invalid input: %s", data[5])
+		}
+
+		detail.UrlPath = data[6]
+		detail.Scheduler = enumor.Scheduler(data[7])
+		session, err := strconv.Atoi(data[8])
```
Comment (chenjr15) [inline cmd/cloud-server/logics/load-balancer/create_url_rule_preview_executor.go:104]: err

File: cmd/cloud-server/logics/load-balancer/create_url_rule_preview_executor.go [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+		curDetail.ValidateResult = append(curDetail.ValidateResult,
+			fmt.Sprintf("URLRule(%s) already exist, and the configuration does not match, port: %d, protocol: %s,"+
+				" domain: %s, url: %s, scheduler: %s, session: %d, healthCheck: %v", rule.ID, listener.Port,
+				listener.Protocol, rule.Domain, rule.URL, rule.Scheduler, rule.SessionExpire, ruleHealthCheck))
+		return nil
+	}
+
+	curDetail.Status.SetExisting()
+	curDetail.ValidateResult = append(curDetail.ValidateResult, fmt.Sprintf("已存在监听器(%s), port: %d, protocol: %s",
+		curDetail.CloudClbID, listener.Port, listener.Protocol))
```
Comment (chenjr15) [inline cmd/cloud-server/logics/load-balancer/create_url_rule_preview_executor.go:241]: 也先改成英文吧

**Review by chenjr15 — APPROVED** [review]

**Pull request merged**